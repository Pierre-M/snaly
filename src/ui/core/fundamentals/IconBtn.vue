<template>
    <button type="button" class="icon-btn" v-haptic :aria-label="label" @click="clickHandler">
        <icon :icon="icon" />
    </button>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";
import Icon from "@/ui/core/fundamentals/Icon.vue";

@Component({
    components: { Icon }
})
export default class IconBtn extends Vue {
    @Prop({ type: String, required: true })
    icon!: string;

    @Prop({ type: String, required: true })
    label!: string;

    clickHandler(e: Event) {
        this.$emit("click", e);
    }
}
</script>

<style lang="scss" scoped>
$icon-btn-inner-space: 10px;

.icon-btn {
    @include boxAlign;
    @include quietButton;
    @include clickable;
    color: inherit;
    font-size: 30px;
    @include withTransition;

    &:active {
        transform: scale(0.8);
    }
}
</style>
