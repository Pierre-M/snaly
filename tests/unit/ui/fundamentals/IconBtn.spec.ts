"use strict";

import { Wrapper } from "@vue/test-utils";
import { shallowMount, triggerDOMEvent } from "../../_utils";
import IconBtn from "@/ui/core/fundamentals/IconBtn.vue";

let wrapper: Wrapper<any>;

describe("IconBtn", () => {
    beforeEach(() => {
        wrapper = shallowMount(IconBtn, {
            propsData: {
                icon: "my-icon"
            },
            slots: {
                default: "My button label"
            }
        });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it("should render given icon", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should emit right events", async () => {
        triggerDOMEvent(wrapper.element, "click");
        expect(wrapper.emitted().click).toBeTruthy();
    });
});
