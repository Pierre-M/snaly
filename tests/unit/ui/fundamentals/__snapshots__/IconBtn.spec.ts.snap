// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconBtn should render given icon 1`] = `
<button type="button" title="My label" aria-label="My label" class="icon-btn flex items-center justify-center p-0 outline-none border-0
        focus:outline-none transform transition duration-200 ease-in-out active:scale-75">
  <icon-stub icon="my-icon" class="text-3xl"></icon-stub>
</button>
`;
